modules/lib/dag.nix

```
@@ -51,66 +50,54 @@ in rec {
   # And the result on error is
   #
   #    {
-  #      cycle = [ {after = ?; name = ?; data = ?} … ];
-  #      loops = [ {after = ?; name = ?; data = ?} … ];
+  #      cycle = [ { after = ?; name = ?; data = ? } … ];
+  #      loops = [ { after = ?; name = ?; data = ? } … ];
   #    }
   #
   # For example
   #
-  #    nix-repl> dagTopoSort {
-  #                a = dagEntryAnywhere "1";
-  #                b = dagEntryAfter ["a" "c"] "2";
-  #                c = dagEntryAfter ["d"] "3";
-  #                d = dagEntryAfter ["b"] "4";
-  #                e = dagEntryAnywhere "5";
+  #    nix-repl> topoSort {
+  #                a = entryAnywhere "1";
+  #                b = entryAfter [ "a" "c" ] "2";
+  #                c = entryAfter [ "d" ] "3";
+  #                d = entryAfter [ "b" ] "4";
+  #                e = entryAnywhere "5";
   #              } == {
   #                cycle = [
-  #                  { after = ["a" "c"]; data = "2"; name = "b"; }
-  #                  { after = ["d"]; data = "3"; name = "c"; }
-  #                  { after = ["b"]; data = "4"; name = "d"; }
+  #                  { after = [ "a" "c" ]; data = "2"; name = "b"; }
+  #                  { after = [ "d" ]; data = "3"; name = "c"; }
+  #                  { after = [ "b" ]; data = "4"; name = "d"; }
   #                ];
   #                loops = [
-  #                  { after = ["a" "c"]; data = "2"; name = "b"; }
+  #                  { after = [ "a" "c" ]; data = "2"; name = "b"; }
   #                ];
-  #              } == {}
+  #              } == { }
   #    true
-  dagTopoSort = dag:
+  topoSort = dag:
     let
       dagBefore = dag: name:
-        mapAttrsToList (n: v: n)
-        (filterAttrs (n: v: any (a: a == name) v.before) dag);
+        builtins.attrNames
+        (filterAttrs (n: v: builtins.elem name v.before) dag);
       normalizedDag = mapAttrs (n: v: {
         name = n;
         data = v.data;
         after = v.after ++ dagBefore dag n;
       }) dag;
-      before = a: b: any (c: a.name == c) b.after;
-      sorted = toposort before (mapAttrsToList (n: v: v) normalizedDag);
+      before = a: b: builtins.elem a.name b.after;
+      sorted = toposort before (builtins.attrValues normalizedDag);
     in if sorted ? result then {
       result = map (v: { inherit (v) name data; }) sorted.result;
     } else
       sorted;
 
   # Applies a function to each element of the given DAG.
-  dagMap = f: dag: mapAttrs (n: v: v // { data = f n v.data; }) dag;
-
-  # Create a DAG entry with no particular dependency information.
-  dagEntryAnywhere = data: {
-    inherit data;
-    before = [ ];
-    after = [ ];
-  };
+  map = f: dag: mapAttrs (n: v: v // { data = f n v.data; }) dag;
```

```suggestion
  map = f: mapAttrs (n: v: v // { data = f n v.data; });
```
This has minimal effect but I think eta reduction is still worth doing while we are at refactoring.

---

Thank you for catching that!